android/app/src/main/java/com/breez/client/plugins/breez/BreezBackup.java

```diff
@@ -0,0 +1,362 @@
+        m_methodChannel.setMethodCallHandler(this);
+
+        new EventChannel(registrar.messenger(), BREEZ_BACKUP_STREAM_NAME).setStreamHandler(this);
+
+        registrar.addActivityResultListener(this);
+
+        m_googleSignInClient = buildGoogleSignInClient();
+        getDriveClients(m_googleSignInClient.silentSignIn());
+    }
+
+    @Override
+    public void onListen(Object args, final EventChannel.EventSink events){
+        m_eventsListener = events;
+    }
+
+    @Override
+    public void onCancel(Object args) {
+        m_eventsListener = null;
+    }
+
+    private GoogleSignInClient buildGoogleSignInClient() {
+        GoogleSignInOptions signInOptions =
+            new GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
+                .requestScopes(Drive.SCOPE_APPFOLDER)
+                .build();
+        return GoogleSignIn.getClient(m_activity, signInOptions);
+    }
+
+    private void getDriveClients(Task<GoogleSignInAccount> task) {
+        Log.i(TAG, "Trying to execute sign in task");
+        task.addOnSuccessListener(
+            googleSignInAccount -> {
+                Log.i(TAG, "Sign in success");
+                m_driveClient = Drive.getDriveClient(m_activity, googleSignInAccount);
+                m_driveResourceClient =
+                        Drive.getDriveResourceClient(m_activity, googleSignInAccount);
+
+            })
+            .addOnFailureListener(
+                e -> {
+                    Log.w(TAG, "Sign in failed", e);
+                });
+    }
+
+    @Override
+    public boolean onActivityResult(int requestCode, int resultCode, Intent data) {
+        if (requestCode == AUTHORIZE_ACTIVITY_REQUEST_CODE) {
+            if (resultCode == RESULT_OK) {
+                getDriveClients(m_googleSignInClient.silentSignIn());
+            }
+            return true;
+        }
+        return false;
+    }
+
+    private void promptToAuthorize() {
+        Intent signInIntent = m_googleSignInClient.getSignInIntent();
+        m_activity.startActivityForResult(signInIntent, AUTHORIZE_ACTIVITY_REQUEST_CODE);
+    }
+
+    @Override
+    public void onMethodCall(MethodCall call, MethodChannel.Result result) {
+        if (call.method.equals("authorize")) {
+            if (m_driveResourceClient == null) {
+                promptToAuthorize();
+            }
+            else {
+                m_result.success(true);
+            }
+        }
+        if (call.method.equals("backup")) {
+            List<String> paths = call.argument("paths");
+            String nodeId = call.argument("nodeId").toString();
+            createNodeIdFolder(paths, nodeId);
+        }
+        if (call.method.equals("listBackupFolders")) {
+            listBackupFolders();
+        }
+        if (call.method.equals("restore")) {
+            if (m_driveResourceClient == null) {
+                promptToAuthorize();
+            }
+            else {
+                m_result = result;
```

Comment:
The nature of the methods in this channel is asynchronous. Therefore saving the m_result as class variable and use it will mix different requests with different results. We should pass the result to the method that is executed to be sure the request/response matches.